src/components/App.js

```diff
@@ -35,183 +38,235 @@ const NotificationsContent = lazy(() => import("./notifications/NotificationsCon
-        const margin = window.location.pathname.endsWith("cbioportal") ? 0 : 26;
-
-        // noinspection HtmlUnknownTarget
-        return <>
-            <Modal title="You have been signed out"
-                   onOk={() => this.openSignInWindow()}
-                   onCancel={() => {
-                       this.clearPingInterval();  // Stop pinging until the user decides to sign in again
-                       this.setState({signedOutModal: false});  // Close the modal
-                       // TODO: Set a new interval at a slower rate
-                   }}
-                   visible={this.state.signedOutModal}>
-                Please <a onClick={() => this.openSignInWindow()}>sign in</a> (uses a popup window) to continue working.
-            </Modal>
-            <Layout style={{minHeight: "100vh"}}>
-                <Suspense fallback={<div />}>
-                    <NotificationDrawer />
-                </Suspense>
-                <SiteHeader />
-                <Layout.Content style={{margin, display: "flex", flexDirection: "column"}}>
-                    <Suspense fallback={<SitePageLoading />}>
-                        <Switch>
-                            <OwnerRoute path={withBasePath("overview")} component={OverviewContent} />
-                            <OwnerRoute path={withBasePath("data/explorer")} component={DataExplorerContent} />
-                            <OwnerRoute path={withBasePath("cbioportal")} component={CBioPortalContent} />
-                            <OwnerRoute path={withBasePath("admin")} component={AdminContent} />
-                            <OwnerRoute path={withBasePath("notifications")} component={NotificationsContent} />
-                            <Redirect from={BASE_PATH} to={withBasePath("overview")} />
-                        </Switch>
-                    </Suspense>
-                </Layout.Content>
-                <SiteFooter />
-            </Layout>
-        </>;
-    }
+    const {accessToken, idTokenContents} = useSelector(state => state.auth);
+    const isAuthenticated = getIsAuthenticated(idTokenContents);
 
-    createEventRelayConnectionIfNecessary() {
-        this.eventRelayConnection = (() => {
-            if (this.eventRelayConnection) {
-                return this.eventRelayConnection;
-            }
+    const eventRelay = useSelector(state => state.services.eventRelay);
+    const eventRelayUrl = eventRelay?.url ?? null;
+    const openIdConfig = useSelector(state => state.openIdConfiguration.data);
 
-            // Don't bother trying to create the event relay connection if the user isn't authenticated
-            if (!this.props.user) return null;
+    const [lastIsAuthenticated, setLastIsAuthenticated] = useState(false);
+
+    const [didPostLoadEffects, setDidPostLoadEffects] = useState(false);
+
+    // TODO: add in localstorage flag to condition to avoid false positives
+    const isInAuthPopup = !!window.opener;
```

Comment:
False positive when coming in from public
![image](https://github.com/bento-platform/bento_web/assets/122313248/5db59dfa-0b2a-4e21-ab17-5b48829bb5cb)